hi-ml-histopathology/src/histopathology/utils/report_utils.py

```diff
@@ -15,8 +15,51 @@
                                 get_tags_from_hyperdrive_run)
 
 
+def run_has_val_and_test_outputs(run: Run) -> bool:
```

Comment:
I think maybe the purpose of this function should be limited to checking whether the run has val and test outputs. Checking whether a test folder is randomly missing feels like an extra responsibility that could be just omitted.

Reply:
Not sure I fully understand what you mean. Currently (with the `DeepMILOutputsHandler`), we expect new experiments to produce `val/test_output.csv` and `test/test_output.csv` (among other output files). Legacy experiments (pre-outputs handler) have only `test_output.csv` (note there's no intermediate folder in this case).

I wrote this function to identify if we're in the first or second case, both of which I'd like to support for now. An exception is thrown if a run doesn't match either pattern.

Reply:
I just don't think this function should be expected to check whether the format is not valid and raise the exception.

Reply:
I find the scope is pretty clear from the docs, and I honestly don't think this function will be heavily used anywhere else.

What would you propose instead? One alternative I see is to switch to just `get_validation_outputs()` and `get_test_outputs()`, which directly try to find the respective files in legacy and new locations, and return `None` if not found.

Reply:
My suggestion was to remove lines 33-36, but if you think it's not critical and it's safer to check if the file tree is not correct, I guess it's fine.